lib/web_console/session.rb

```diff
@@ -59,11 +59,44 @@ def eval(input)
     #
     # Returns nothing.
     def switch_binding_to(index)
-      @evaluator = Evaluator.new(@bindings[index.to_i])
+      @evaluator = Evaluator.new(@current_binding = @bindings[index.to_i])
+    end
+
+    # Returns context of the current binding
+    def context(obj)
+      ( object_name?(obj) ? context_of(obj) : global_context ).flatten
     end
 
     private
 
+      def object_name?(s)
+        s.is_a?(String) && !s.empty? && !s.match(/[^a-zA-Z0-9\@\$\.\:]/)
+      end
+
+      def context_eval(cmd)
```

Comment:
It seems that a `Context` object is emerging here. Do you wanna move all this code to `lib/web_console/context.rb` in a `WebConsole::Context` class. We can feed it the command and have the current global context as state.